docs/installation.rst

```diff
@@ -19,13 +19,4 @@ latest GitHub main::
 
     $ pip install git+https://github.com/zarr-developers/zarr-python.git
 
-To work with Zarr source code in development, install from GitHub::
-
-    $ git clone --recursive https://github.com/zarr-developers/zarr-python.git
-    $ cd zarr-python
-    $ python -m pip install -e .
-
```

Comment:
This snippet was missing the pip install for dev requirements files. I realized the contributing.rst docs were much more complete and I propose just linking to those so theres a single source of truth